### PR TITLE
Catch unchecked exceptions in Mesos driver.

### PR DESCRIPTION
**Pull request by jeffchao**
### Context

Helps verify races related to Mesos driver initialization during leader election.

### Checklist

- [x] `./gradlew build` compiles code correctly
- [x] Added new tests where applicable
- [x] `./gradlew test` passes all tests
- [x] Extended README or added javadocs where applicable
- [x] Added copyright headers for new files from `CONTRIBUTING.md`

### CODE REVIEW

File: server/src/main/java/io/mantisrx/server/master/mesos/VirtualMachineMasterServiceMesosImpl.java [hunk 1]
```diff
@@ -392,15 +392,22 @@ public void killTask(final WorkerId workerId) {
...
+                    TaskID.newBuilder()
+                            .setValue(taskIdString)
+                            .build());
+            logger.info("Kill status = " + status);
+            switch (status) {
+                case DRIVER_ABORTED:
+                case DRIVER_STOPPED:
+                    logger.error("Unexpected to see Mesos driver status of " + status + " from kill task request. Committing suicide!");
+                    System.exit(2);
+            }
```
Comment (jeffchao) [inline server/src/main/java/io/mantisrx/server/master/mesos/VirtualMachineMasterServiceMesosImpl.java:407]: Diff is bad. these lines are functionally unchanged.

File: server/src/main/java/io/mantisrx/server/master/mesos/VirtualMachineMasterServiceMesosImpl.java [hunk 2]
```diff
@@ -392,15 +392,22 @@ public void killTask(final WorkerId workerId) {
...
+        } catch (RuntimeException e) {
+            // IllegalStateException from no mesosDriver's addVMLeaseAction or NPE from mesosDriver.get() being null.
+            logger.error("Unexpected to see Mesos driver not initialized", e);
             System.exit(2);
```
Comment (jeffchao) [inline server/src/main/java/io/mantisrx/server/master/mesos/VirtualMachineMasterServiceMesosImpl.java:411]: Added this line.

**Review by neerajrj — APPROVED** [review]
looks good to me

**Comment by codyrioux** [comment]
LGTM as well.

**Review by codyrioux — APPROVED** [review]
LGTM

**Comment by jeffchao** [comment]
Thanks all!

**Pull request merged**